File: echo-guardian-production-foundation-v1.0/native/ios/EchoGuardianKit/Sources/EchoGuardianKit/Policy/NativePolicyStatus.swift
import Foundation

public struct NativePolicyStatus: Codable, Equatable {
    public let monitoringEnabled: Bool
    public let privateSpaceMonitoringEnabled: Bool
    public let authorityLabel: String
    public let cloudRequiredForCoreLoop: Bool
    public let silentTelemetryEnabled: Bool
    public let plainLanguage: String

    public init(monitoringEnabled: Bool, privateSpaceMonitoringEnabled: Bool, authorityLabel: String, cloudRequiredForCoreLoop: Bool = false, silentTelemetryEnabled: Bool = false) {
        precondition(privateSpaceMonitoringEnabled == false, "Private-space monitoring is blocked by default in v0.5")
        precondition(cloudRequiredForCoreLoop == false, "Cloud dependency is blocked for the core loop")
        precondition(silentTelemetryEnabled == false, "Silent telemetry is blocked")
        self.monitoringEnabled = monitoringEnabled
        self.privateSpaceMonitoringEnabled = privateSpaceMonitoringEnabled
        self.authorityLabel = authorityLabel
        self.cloudRequiredForCoreLoop = cloudRequiredForCoreLoop
        self.silentTelemetryEnabled = silentTelemetryEnabled
        self.plainLanguage = monitoringEnabled ? "Echo Guardian monitoring is on. You can see what is being watched." : "Echo Guardian monitoring is off."
    }
}

File: echo-guardian-production-foundation-v1.0/native/android/echo-guardian-core/src/main/java/org/echoguardian/policy/NativePolicyStatus.kt
package org.echoguardian.policy

data class NativePolicyStatus(
    val monitoringEnabled: Boolean,
    val privateSpaceMonitoringEnabled: Boolean,
    val authorityLabel: String,
    val cloudRequiredForCoreLoop: Boolean = false,
    val silentTelemetryEnabled: Boolean = false,
) {
    init {
        require(!privateSpaceMonitoringEnabled) { "Private-space monitoring is blocked by default in v0.5" }
        require(!cloudRequiredForCoreLoop) { "Cloud dependency is blocked for the core loop" }
        require(!silentTelemetryEnabled) { "Silent telemetry is blocked" }
    }

    val plainLanguage: String = if (monitoringEnabled) {
        "Echo Guardian monitoring is on. You can see what is being watched."
    } else {
        "Echo Guardian monitoring is off."
    }
}
